transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -718,12 +752,37 @@ protected void doDisconnect() throws Exception {
 
     @Override
     protected void doClose() throws Exception {
+        Queue<AbstractMap.SimpleEntry<Object, ChannelPromise>> queue = tempInboundMessages();
+        if (isNotEmpty(queue)) {
+            queue.clear();
```

Comment:
This is not enough... You will need to release the message as well and fail the promise.

```
ClosedChannelException exception = null;
for (;;) {
    AbstractMap.SimpleEntry<Object, ChannelPromise> entry = queue.poll();
    if (entry == null) {
        break;
    }
    ReferenceCountedUtil.release(entry.getKey());
    ChannelPromise promise = entry.getValue();
    if (promise != null) {
        if (exception == null) {
            exception = new ClosedChannelException();
        }
        promise.tryFailure(exception);
    }
}
```

Reply:
My fault...
Fixed ...